org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -219,59 +220,69 @@ public JavaProtocolExtensions getJavaExtensions() {
 	@Override
 	public void didChangeConfiguration(DidChangeConfigurationParams params) {
 		logInfo(">> workspace/didChangeConfiguration");
+		JobHelpers.waitForInitializeJobs();
+		if (preferenceManager.getClientPreferences().isWorkspaceSymbolDynamicRegistered()) {
+			JavaLanguageServerPlugin.getInstance().registerCapability(Preferences.WORKSPACE_SYMBOL_ID, Preferences.WORKSPACE_SYMBOL);
+		}
+		if (preferenceManager.getClientPreferences().isDocumentSymbolDynamicRegistered()) {
+			JavaLanguageServerPlugin.getInstance().registerCapability(Preferences.DOCUMENT_SYMBOL_ID, Preferences.DOCUMENT_SYMBOL);
+		}
```

Comment:
looks like both symbols related capabilities need to be registered ASAP, i.e mustn't wait for the init job to finish, or else you get no workspace wide errors/warnings (try with https://github.com/spring-projects/spring-amqp )

Reply:
In my opinion, the issue happens because o.e.core.resources plugin hasn't been started correctly and autobuilding is set to off. I have created a new commit that would also need to fix https://github.com/redhat-developer/vscode-java/issues/380

Reply:
Starting the o.e.core.resources plugin does indeed seem to fix the missing diagnostics problem, but it doesn't fix redhat-developer/vscode-java#380

Reply:
@snjeza so it almost works, except for https://github.com/eclipse/eclipse.jdt.ls/tree/master/org.eclipse.jdt.ls.tests/projects/eclipse/hello (after adding unused imports on all java files). On restart we only receive diagnostics for the opened file(s).
Using the previous version of the server, all diagnostics are reported.